adapters/httpapi_compact.c

```diff
@@ -66,9 +66,10 @@ typedef struct HTTP_HANDLE_DATA_TAG
 
 static HTTPAPI_RESULT OpenXIOConnection(HTTP_HANDLE_DATA* http_instance);
 
-/*the following function does the same as sscanf(pos2, "%d", &sec)*/
+/*Read a decimal from a char[], returns a decimal and the remaining part which cannot be converted to a number.*/
+/*the following function does the same as sscanf(pos2, "%d %s", &sec, remaining)*/
 /*this function only exists because some of platforms do not have sscanf. */
-static int ParseStringToDecimal(const char *src, int* dst)
+static int ParseStringToDecimalAndRemaining(const char *src, int* dst, char* remaining)
```

Comment:
need add one more parameter telling length of the buffer, otherwise there could be buffer over run